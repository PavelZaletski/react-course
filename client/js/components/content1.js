import React from 'react';

export default () => (
    <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eu lacinia tellus. Sed et quam risus. Phasellus sit amet </p>
);