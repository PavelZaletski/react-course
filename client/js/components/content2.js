import React from 'react';

export default class Content2 extends React.PureComponent {
    render() {
        return (
            <p>Nulla rhoncus tempus erat nec facilisis. Duis quis purus ligula. Lorem ipsum dolor sit amet</p>
        );
    }
}