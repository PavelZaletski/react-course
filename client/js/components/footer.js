import React from 'react';

export default function() {
    return (
        <footer>Netflixroulette</footer>
    );
}