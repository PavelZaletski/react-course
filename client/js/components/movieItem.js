import React from 'react';

export default function ({ item }) {
    const Genres = item.genres.map(genre => <span key={genre}>{genre}</span>);

    return (
        <div className="movie-item">
            <img className="movie-item__img" src={item.poster_path} />
            <div className="movie-item__description">
                <h4 className="movie-item__title">{item.title}</h4>
                <span className="movie-item__release-date">{new Date(item.release_date).getFullYear()}</span>
            </div>
            <div className="movie-item__genres">{Genres}</div>
        </div>
    );
}