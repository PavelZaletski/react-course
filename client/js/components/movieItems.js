import React from 'react';
import { TopContainer } from './topContainer';
import MovieItem from './movieItem';
import { connect } from 'react-redux';
import { fetchMovieById, fetchMoviesByGenres } from '../actions/movies-actions';

class MovieItems extends React.Component {
    render() {
        const { relatedMovies } = this.props;
        const Items = relatedMovies && relatedMovies.map(item => <MovieItem item={item} key={item.id} />);

        return (
            <div className="movies-list">
                <div className="movies-list__header">
                    <span>Films by {this.props.genre} genre</span>
                    
                </div>
                <div className="movies-list__items">
                    {Items}
                </div>
            </div>
        );
    }
}

const mapStateToProps = (store) => ({
    relatedMovies: store.movies.relatedMovies
});

export default connect(mapStateToProps)(MovieItems);