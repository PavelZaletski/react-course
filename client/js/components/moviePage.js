import React from 'react';
import { TopContainer } from './topContainer';

export class MoviePage extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        const { movie } = this.props;

        return (
            movie ?
                <TopContainer>
                    <div className="movie-page__info">
                        <div className="movie-page__img-wrapper">
                            <img src={movie.poster_path} className="movie-page__img"/>
                        </div>
                        <div className="movie-page__description">
                            <h2 className="movie-page__title">{movie.title}</h2>
                            <div className="movie-page__tagline"> {movie.tagline}</div>
                            <span className="movie-page__year">{new Date(movie.release_date).getFullYear()}</span>
                            { movie.runtime && <span className="movie-page__runtime">{`${movie.runtime} min`}</span> }
                            <div className="movie-page__overview"> {movie.overview}</div>
                        </div>
                    </div>
                </TopContainer>
            :
            null
        );
    }
}