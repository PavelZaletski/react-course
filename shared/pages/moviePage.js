// @flow

import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import TopContainer from '../components/topContainer';
import { fetchMovieById, fetchMoviesByGenres } from '../actions/movies-actions';
import MovieItems from '../components/movieItems';
import { Movie } from '../types/movie';

type Props = {
  match: any,
  movie: Movie,
  fetchMovieById: (value: number) => void,
  fetchMoviesByGenres: (value: string) => void,
  errorMessage: string,
  fetched: boolean
};

export class MoviePageComponent extends React.Component<Props> {
  constructor(props) {
    super(props);
    this.props.fetchMovieById(this.props.match.params.id);
  }

  render() {
    const { movie, errorMessage } = this.props;
    const genre = movie && movie.genres && movie.genres[0];

    return (
      movie ?
            <div>
                {errorMessage && <div>{errorMessage}</div>}
                <TopContainer>
                    <div className="movie-page__info">
                        <div className="movie-page__img-wrapper">
                            <img src={movie.poster_path} className="movie-page__img" />
                        </div>
                        <div className="movie-page__description">
                            <h2 className="movie-page__title">{movie.title}</h2>
                            <div className="movie-page__tagline"> {movie.tagline}</div>
                            <span className="movie-page__year">
                              {new Date(movie.release_date).getFullYear().toString()}
                            </span>
                            {movie.runtime && <span className="movie-page__runtime">{`${movie.runtime} min`}</span>}
                            <div className="movie-page__overview"> {movie.overview}</div>
                            <Link className="home-btn" to='/' >Search</Link>
                        </div>
                    </div>
                </TopContainer>

                {genre && <MovieItems genre={genre}/> }
            </div>
        :
        null
    );
  }
}

const mapStateToProps = store => ({
  movie: store.movies.selectedMovie,
  fetched: store.movies.selectedMovieFetched,
  errorMessage: store.movies.errorMessage,
});

const mapDispatchToProps = dispatch => ({
  fetchMovieById: (id) => {
    dispatch(fetchMovieById(id));
  },

  fetchMoviesByGenres: (genres) => {
    dispatch(fetchMoviesByGenres(genres));
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(MoviePageComponent);

