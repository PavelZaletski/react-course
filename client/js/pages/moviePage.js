import React from 'react';
import { TopContainer } from '../components/topContainer';
import { connect } from 'react-redux';
import { fetchMovieById, fetchMoviesByGenres } from '../actions/movies-actions';
import MovieItems from '../components/movieItems';

export class MoviePageComponent extends React.Component {
    componentDidMount() {
        this.props.fetchMovieById(this.props.match.params.id);
    }

    render() {
        const { movie, errorMessage } = this.props;
        let genre = movie && movie.genres && movie.genres[0];

        return (
            movie ?
            <div>
                {errorMessage && <div>{errorMessage}</div>}
                <TopContainer>
                    <div className="movie-page__info">
                        <div className="movie-page__img-wrapper">
                            <img src={movie.poster_path} className="movie-page__img" />
                        </div>
                        <div className="movie-page__description">
                            <h2 className="movie-page__title">{movie.title}</h2>
                            <div className="movie-page__tagline"> {movie.tagline}</div>
                            <span className="movie-page__year">{new Date(movie.release_date).getFullYear()}</span>
                            {movie.runtime && <span className="movie-page__runtime">{`${movie.runtime} min`}</span>}
                            <div className="movie-page__overview"> {movie.overview}</div>
                        </div>
                    </div>
                </TopContainer>

                <MovieItems genre={genre}/>
            </div>
            :
            null
        );
    }
}

const mapStateToProps = (store) => ({
    movie: store.movies.selectedMovie,
    fetched: store.movies.selectedMovieFetched,
    errorMessage: store.movies.errorMessage
});

const mapDispatchToProps = (dispatch) => ({
    fetchMovieById: (id) => {
        dispatch(fetchMovieById(id))
    },

    fetchMoviesByGenres: (genres) => {
        dispatch(fetchMoviesByGenres(genres))
    }
});

export default connect(mapStateToProps, mapDispatchToProps)(MoviePageComponent);

