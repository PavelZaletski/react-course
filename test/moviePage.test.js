import renderer from 'react-test-renderer';
import Enzyme, { shallow, mount } from 'enzyme';
import React from 'react';
import { MoviePage } from '../client/js/components/moviePage';

describe('MoviePage', () => {
    it('should match snapshot', () => {
        let component = renderer.create(<MoviePage />).toJSON();
        expect(component).toMatchSnapshot();
    });

    it('should match snapshot', () => {
        const item = {
            poster_path: 'test',
            title: 'title',
            release_date: '2015-01-01',
            tagline: 'some description...',
            overview: 'lorem ipsum dolor'
        };

        let component = renderer.create(<MoviePage item={item} />).toJSON();
        expect(component).toMatchSnapshot();

        item.runtime = 100;

        component = renderer.create(<MoviePage item={item} />).toJSON();
        expect(component).toMatchSnapshot();
    });
});