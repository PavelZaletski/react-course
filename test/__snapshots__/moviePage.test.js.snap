// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePage should match snapshot 1`] = `null`;

exports[`MoviePage should match snapshot 2`] = `null`;

exports[`MoviePage should match snapshot 3`] = `null`;
